File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    const answer = [];
    
    const record = {};
    players.forEach((item, idx) => record[item] = idx);
    
    callings.forEach((item) => {
        const currentIdx = record[item];
        const prevIdx = record[item] - 1;
        
        players[currentIdx] = players[prevIdx];
        players[prevIdx] = item;
        
        record[item] = prevIdx;
        record[players[currentIdx]] = currentIdx;
    })
    
    return players;
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    obj = {}
    n = len(players)
    
    for idx, player in enumerate(players):
        obj[player] = idx
    
    for call in callings:
        now = obj[call] # 3
        prev = now - 1 # 2
        
        players[now], players[prev] = players[prev], players[now]
        
        obj[call] -= 1
        obj[players[now]] += 1
    
    return players